### PR TITLE
feat(orc8r): optional agw device on register from northbound endpoints

### PR DESCRIPTION
**Pull request by reginawang3495**
## Summary
- make `device` optional on POST register from northbound endpoints
- make `registrationInfo` returned on GET gateway from northbound endpoints if device doesn't exist

## Test Plan
- fix up tests and run orc8r tests
- manually validate that endpoints work

## Additional Information

- [ ] This change is backwards-breaking

### CODE REVIEW

File: orc8r/cloud/go/services/bootstrapper/servicers/registration/cloud_registration.go [hunk 1]
```diff
@@ -53,7 +56,7 @@ func (c *cloudRegistrationServicer) GetToken(ctx context.Context, request *proto
 		glog.V(2).Infof("could not get tokenInfo for networkID %v and logicalID %v: %v", networkId, logicalId, err)
 	}
 
-	refresh := request.Refresh || tokenInfo == nil || IsExpired(tokenInfo)
+	refresh := (c.refreshOnRequest && request.Refresh) || tokenInfo == nil || IsExpired(tokenInfo)
```
Comment (reginawang3495) [inline orc8r/cloud/go/services/bootstrapper/servicers/registration/cloud_registration.go:59]: @hcgatewood i feel like we could just remove `request.Refresh` -- it kinda makes it confusing

File: orc8r/cloud/go/services/orchestrator/obsidian/models/conversion.go [hunk 2]
```diff
@@ -144,14 +146,19 @@ func (m *MagmadGateway) GetMagmadGateway() *MagmadGateway {
...
+
 	return []configurator.EntityWriteOperation{
 		configurator.NetworkEntity{
 			Type:        orc8r.MagmadGatewayType,
 			Key:         string(m.ID),
 			Name:        string(m.Name),
 			Description: string(m.Description),
 			Config:      m.Magmad,
-			PhysicalID:  m.Device.HardwareID,
+			PhysicalID:  physicalID,
```
Comment (reginawang3495) [inline orc8r/cloud/go/services/orchestrator/obsidian/models/conversion.go:161]: input requested on whether this will work as expected

File: feg/cloud/go/services/feg/obsidian/models/swagger.v1.yml [hunk 3]
```diff
@@ -345,7 +345,7 @@ paths:
 
   /feg/{network_id}/gateways:
     get:
-      summary: List all federation gateways for a federation network
+      summary: List all gateways for a federation network
```
Comment (andreilee) [inline feg/cloud/go/services/feg/obsidian/models/swagger.v1.yml:348]: Revert this change - our definitions of `federation network`, `federated lte network`, `lte network`, etc. are a bit confusing, and this additional context helps.

File: orc8r/cloud/go/models/swagger-common.yml [hunk 4]
```diff
@@ -151,3 +153,23 @@ definitions:
       message:
         type: string
         example: Error string
+  registration_info:
```
Comment (andreilee) [inline orc8r/cloud/go/models/swagger-common.yml:156]: add a helpful description please

File: orc8r/cloud/go/models/swagger-common.yml [hunk 5]
```diff
@@ -151,3 +153,23 @@ definitions:
...
+        minLength: 4
+        example: reg_a9vnap30fN0anrfjVneB
+      root_ca:
+        description: root_ca is a certificate that access gateways (AGW) can use to handshake and communicate with the stated orc8r
+        type: string
+        example: =example rootCA.pem=
+      domain_name:
+        description: domain_name is the domain name where the stated orc8r can be accessed
+        type: string
+        example: example.com
```
Comment (andreilee) [inline orc8r/cloud/go/models/swagger-common.yml:175]: This seems a little strange to me - wouldn't this be the same for every gateway? Why is this configurable per gateway?

File: orc8r/cloud/go/services/orchestrator/obsidian/models/conversion.go [hunk 6]
```diff
@@ -239,10 +246,47 @@ func (m *MagmadGateway) FromBackendModels(ent configurator.NetworkEntity, device
 	if err == nil {
 		m.Tier = TierID(tierTK.Key)
 	}
+	err = PopulateRegistrationInfo(context.Background(), m, ent.NetworkID)
+	// ignore err and continue returning the rest of the gateway
+	glog.V(2).Infof("failed to populate registration info because %v+", err)
```
Comment (andreilee) [inline orc8r/cloud/go/services/orchestrator/obsidian/models/conversion.go]: Shouldn't this log only if `err` is nil?

File: feg/cloud/go/services/feg/obsidian/models/swagger.v1.yml (hunk 3 above)
Comment (reginawang3495) [reply feg/cloud/go/services/feg/obsidian/models/swagger.v1.yml:348]: the reason i have this is to adhere to the `POST` register gateway naming for all the other networks
```
                    - “List all gateways for a Wifi network”
                    - “List all gateways for a carrier wifi network”
                    - “List all federation gateways for a federation network” [ RENAME ]
                    - “List all gateways for a network”
                    - “List all gateways for an LTE network”
  ```

File: orc8r/cloud/go/models/swagger-common.yml (hunk 5 above)
Comment (reginawang3495) [reply orc8r/cloud/go/models/swagger-common.yml:175]: it is returned by the orc8r to make registering from the device faster -- not configurable

File: orc8r/cloud/go/services/orchestrator/obsidian/models/conversion.go (hunk 6 above)
Comment (reginawang3495) [reply orc8r/cloud/go/services/orchestrator/obsidian/models/conversion.go]: yes nice catch ty

File: orc8r/cloud/go/models/swagger-common.yml (hunk 4 above)
Comment (reginawang3495) [reply orc8r/cloud/go/models/swagger-common.yml:156]: the other ones don't have a description and im slightly on a timeline if you wouldn't mind leaving as is (since generating takes a while)

**Review by hcgatewood — APPROVED** [review]

**Review by themarwhal — APPROVED** [review]
